MediaGalleryUi/Plugin/MediaGalleryIndexerTrigger.php

```diff
@@ -42,8 +42,8 @@ public function __construct(
      */
     public function afterSave(Value $config, Value $result): Value
     {
-        $isMediaGallery = $result->getPath() == self::MEDIA_GALLERY_CONFIG_VALUE;
-        if ($isMediaGallery && $result->isValueChanged() && $result->getValue() == self::MEDIA_GALLERY_ENABLED_VALUE) {
+        $isMediaGallery = $result->getPath() === self::MEDIA_GALLERY_CONFIG_VALUE;
+        if ($isMediaGallery && $result->isValueChanged() && (int) $result->getValue() === self::MEDIA_GALLERY_ENABLED_VALUE) {
```

Comment:
I personaly would remove a variable and put all checks in one if statement like this
```suggestion
        if ($result->getPath() === self::MEDIA_GALLERY_CONFIG_VALUE 
            && $result->isValueChanged() 
            && (int)$result->getValue() === self::MEDIA_GALLERY_ENABLED_VALUE
        ) {
```

Reply:
@hws47a thank you for your patient to details! I appreciate this.

I made some changes. I moved `isValueChanged` from the if condition. I made this based on the next conclusion: identifying that the path is a media config and that the value changed looks for me as a stand-alone action. So good to keep them separate. What do you think?

Reply:
I think introducing variables for not complex parts of the code makes it slower to read.

Also for the current case with having a second variable, you call `isValueChanged` even then it's not needed, I mean if `$isMediaGallery` will be false.

Reply:
Agree. Will provide changes in a few minutes.

Reply:
@hws47a thank you for details. I made the recommended changes. Let's wait for the static tests finished and review :)